src/Sort.ts

```diff
@@ -101,9 +106,33 @@ export class Sort {
 
     // Currently sorts by first tag found only.
     private static compareByTag(a: Task, b: Task): -1 | 0 | 1 {
-        if (a.tags[0] < b.tags[0]) {
+        // If no tags then assume they are equal.
+        if (a.tags.length === 0 && b.tags.length === 0) {
+            return 0;
+        } else if (a.tags.length === 0) {
+            // a is less than b
+            return 1;
+        } else if (b.tags.length === 0) {
+            // b is less than a
+            return -1;
+        }
+
+        // If the tag collection is smaller than the instance
+        // used to compare then they are just equal.
+        if (
+            Sort.tagPropertyInstance >= a.tags.length ||
+            Sort.tagPropertyInstance >= b.tags.length
+        ) {
+            return 0;
+        }
+
+        if (
+            a.tags[Sort.tagPropertyInstance] < b.tags[Sort.tagPropertyInstance]
+        ) {
             return -1;
-        } else if (a.tags[0] > b.tags[0]) {
+        } else if (
+            a.tags[Sort.tagPropertyInstance] > b.tags[Sort.tagPropertyInstance]
+        ) {
```

Comment:
Thanks. Also, feel free to alternatively log it as a know issue and deal with it as a separate PR...

Reply:
Sound like this would also be a potential issue with multiple tags. Make sure spaces are not allowed as part of a tag character. That should prevent this from happening. As far as I know, a space is not allowed as part of an Obsidian tag.

Reply:
It's sorting by the first "tag", right? I think the quickest fix to accompany relevant use cases is to ensure that the tag is preceded by a space. What do you think?

Reply:
Fixed the issue, it will only accept tags that have a space character before the # symbol or a start of line. Updated comments to reflect this as well.